src/lib.rs

```diff
@@ -13,6 +13,12 @@
 
 #![cfg_attr(feature = "nightly", feature(drain))]
 
+// optional serde support
+#![cfg_attr(feature = "serde_macros", feature(const_fn, custom_derive, plugin))]
```

Comment:
I just used the example here (under the section about using both):

https://github.com/serde-rs/serde#using-serde-with-stable-rust-syntex-and-serde_codegen

It doesn't really matter to me, and I'm struggling to think of a scenario where you'd compile with serde support but without those features (unless custom_derive makes it to stable? but I assume other changes would be needed to make it work).

If that's what you'd prefer, I'm happy to make the change.